### PR TITLE
Add .well-known/change-password url

### LINKED ISSUE #3991: /.well-known/change-password
**Is your feature request related to a problem? Please describe.**
There is a standard for apps aware of credential management, specifically for asking a site to send you to the place to change the current user's password: https://wicg.github.io/change-password-url/index.html

**Describe the solution you'd like**
Redirect `/.well-known/change-password` to [`/profile/edit?section=password
`](https://www.worldcubeassociation.org/profile/edit?section=password
)

### PR DESCRIPTION
**Pull request by UnsolvedCypher**
Closes #3991

### CODE REVIEW

File: WcaOnRails/config/routes.rb [hunk 1]
```diff
@@ -186,6 +186,8 @@
   get '/relations' => 'relations#index'
   get '/relation' => 'relations#relation'
 
+  get '/.well-known/.change-password' => redirect('/profile/edit?section=password', status: 302)
```
Comment (jonatanklosko) [inline WcaOnRails/config/routes.rb]: Nit: we don't want the dot before `change-password`.

**Comment by jonatanklosko** [comment]
Thanks!

**Comment by lgarron** [comment]
Hmm, two things:

- https://www.worldcubeassociation.org/profile/edit?section=password still gives me a 404. :-( I tried deploying `master` just in case, and still not.
- Visiting https://www.worldcubeassociation.org/profile/edit?section=password prompted for a password to access sensitive data. But it actually selected the **2FA tab** during that password prompt, then redirected me to the general edit page

Nothing too terrible, just a bit confusing.

**Comment by jonatanklosko** [comment]
Interesting:

- I see a request to `/.well-known/change-password` in the Nginx log, but not in the Rails log, so it probably has to do with the configuration, cannot dive into that now
- cc-ing @viroulep

**Comment by UnsolvedCypher** [comment]
I think this may be related to the problem: https://stackoverflow.com/questions/20496963/avoid-nginx-decoding-query-parameters-on-proxy-pass-equivalent-to-allowencodeds

**Comment by viroulep** [comment]
>Visiting https://www.worldcubeassociation.org/profile/edit?section=password prompted for a password to access sensitive data. But it actually selected the 2FA tab during that password prompt, then redirected me to the general edit page

It's probably because I relied on our tab-switching mechanism which relies on the value of the section parameter.
One workaround I could see: we could hook into the js for the tab-switching mechanism, then look for the div with id matching the section, and if not found then show the password/2FA confirmation without changing the section value.

**Comment by jonatanklosko** [comment]
Actually there is no sign of that request in the Rails log, there is however an Nginx error:

```
2020/02/26 14:21:21 [error] 21567#0: *41199388 open() "/home/cubing/worldcubeassociation.org/WcaOnRails/public/.well-known/change-password" failed (2: No such file or directory), client: [...], server: www.worldcubeassociation.org, request: "GET /.well-known/change-password HTTP/1.1", host: "www.worldcubeassociation.org"
```

Looks like Nginx assumes this is a static file and looks it up in the `root` folder and then fails (instead of passing it to the Rails app). Looking at the Nginx config I don't see why that would be the case, perhaps it just assumes it is a file because of the leading dot, but that's just a wild guess.

If someone has any idea, would be happy to hear. I think it may make sense to handle the redirect on Nginx level instead of spending too much time investigating that though.

**Comment by FinnIckler** [comment]
I just came across this issue when setting up OIDC which also comes with .well-known paths.
The issue is indeed this line in the nginx conf
  ```
# Deny access to dotfiles and dotfolders.
  location ~ /\. {
    deny all;
  }
```
We are getting rid of the nginx soon anyway so this will work in the future, but it still good to know that this is a dead route in our codebase at the moment!

**Pull request merged**